Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -846,6 +847,330 @@ TEST_F(AssetProcessorIntermediateAssetTests, IntermediateAsset_SourceNoLongerEmi
+    //      they are depending on is missing, and the intermediate asset won't be available yet due to the first point.
+    //  then, later during processing, after the intermediate asset is created and the job is run on it
+    //      The asset processor matches the intermediate asset to the previously missing dependency, and re-queues the
+    //      source asset with the dependency.
+    //  finally, the initially run job runs a second time, because it was re-queued.
+    //      This is why FBX files eventually process without failure, when asset processor hits idle it has zero failures.
+    //
+    // This is not desired behavior long term. However, this test is a step toward the solution. It is a regression test
+    // for that particular scenario. When this job ordering situation is resolved, then this test will need to be updated
+    // to match the new behavior.
+
+    // Source Asset A emits an intermediate asset, A1
+    // Source Asset B has a source dependency to the path of A1
+    // Source Asset B has processed before Source Asset A, so A1 is not yet available.
+    // Source Asset B emits warnings because the source dependency does not exist.
+    // Source Asset A processes, intermediate asset A1 becomes available
+    // Intermediate Asset A1 processes
+    // Source Asset B is added to the reprocessing queue, because A1 resolves the missing dependency.
+    // Source Asset B is re-processed because the dependency is now resolved.
+    // Source Asset B processes without warnings, because A1 now exists.
+
+    QMetaObject::invokeMethod(
+        m_assetProcessorManager.get(), "AssessAddedFile", Qt::QueuedConnection, Q_ARG(QString, m_firstFilePath.c_str()));
+    QMetaObject::invokeMethod(
+        m_assetProcessorManager.get(), "AssessAddedFile", Qt::QueuedConnection, Q_ARG(QString, m_secondFilePath.c_str()));
+
+    QCoreApplication::processEvents();
+    m_fileCompiled = false;
+    m_fileFailed = false;
+    RunFile(2, 2);
+
+    // Queue both jobs at the same time, so that RCQueueSortModel.cpp will order these jobs and run them in order.
+    // TODO: Note that the PC job (asset B) will always run before the Common job (asset A) because the system
+    // purposely prioritizes host platform jobs first.
+    ASSERT_EQ(m_jobDetailsList.size(), 2);
+    m_rc->JobSubmitted(m_jobDetailsList[0]);
+    m_rc->JobSubmitted(m_jobDetailsList[1]);
+    m_jobDetailsList.clear();
+
+    UnitTests::JobSignalReceiver receiver;
+
+    // Processing asset B will produce several warnings right now (listed at the end of this block, where they are checked).
+    m_errorChecker.Begin();
+
+    // Process the first asset in the queue, which will be the job for B, because it's a PC job, which takes priority over the Common platform job in the queue.
+    QCoreApplication::processEvents();
+    receiver.WaitForFinish();
+    QCoreApplication::processEvents();
+    QCoreApplication::processEvents();
```

Comment:
I know its a bit of a pain but if you can put comments on what function should be running during each processEvents, it helps keep track of what's happening when reading the test and also with knowing where to look to debug.

Reply:
I went through and commented them, as well as collapsed one set into a function.